packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/pattern_item.dart

```diff
@@ -4,32 +4,57 @@
 
 import 'package:flutter/foundation.dart' show immutable;
 
+/// Enumeration of types of pattern items.
+enum PatternItemType {
+  /// A dot used in the stroke pattern for a [Polyline].
+  dot,
+
+  /// A dash used in the stroke pattern for a [Polyline].
+  dash,
+
+  /// A gap used in the stroke pattern for a [Polyline].
+  gap,
+}
+
+String _patternItemTypeToJson(PatternItemType itemType) => switch (itemType) {
+      PatternItemType.dot => 'dot',
+      PatternItemType.dash => 'dash',
+      PatternItemType.gap => 'gap',
+    };
+
 /// Item used in the stroke pattern for a Polyline.
 @immutable
 class PatternItem {
-  const PatternItem._(this._json);
+  const PatternItem._(this.patternItemType, [this.length]);
 
   /// A dot used in the stroke pattern for a [Polyline].
-  static const PatternItem dot = PatternItem._(<Object>['dot']);
+  static const PatternItem dot = PatternItem._(PatternItemType.dot);
 
   /// A dash used in the stroke pattern for a [Polyline].
   ///
   /// [length] has to be non-negative.
   static PatternItem dash(double length) {
     assert(length >= 0.0);
-    return PatternItem._(<Object>['dash', length]);
+    return PatternItem._(PatternItemType.dash, length);
   }
 
   /// A gap used in the stroke pattern for a [Polyline].
   ///
   /// [length] has to be non-negative.
   static PatternItem gap(double length) {
     assert(length >= 0.0);
-    return PatternItem._(<Object>['gap', length]);
+    return PatternItem._(PatternItemType.gap, length);
   }
 
-  final Object _json;
+  /// The type of rendering used for an item in a pattern.
+  final PatternItemType patternItemType;
+
+  /// The length in pixels of a dash or gap.
+  final double? length;
```

Comment:
Thinking about the future usage here again, don't we need different subclasses for whether or not there's a required length? With this version, consuming code is going to have to force-unwrap `length` for certain pattern types, which type safety was lost along the way (since length was a non-optional and non-nullable param when the pattern was created.)